flink-runtime/src/main/scala/org/apache/flink/runtime/messages/JobManagerMessages.scala

```diff
@@ -95,6 +95,14 @@ object JobManagerMessages {
   case class CancelJob(jobID: JobID) extends RequiresLeaderSessionID
 
   /**
+   * Stops a (streaming) job with the given [[jobID]] at the JobManager. The result of
+   * stopping is sent back to the sender as a [[StoppingResponse]] message.
+   *
+   * @param jobID
+   */
+  case class StopJob(jobID: JobID)
```

Comment:
`StopJob` should extend `RequiresLeaderSessionID` in order to cope with changing leaders.